packages/iframe-coordinator/src/elements/frame-router.ts

```diff
@@ -273,6 +273,6 @@ export default class FrameRouterElement extends HTMLElement {
       return hostUrlObject.href;
     }
     const trimmedUrl = stripTrailingSlash(hostUrl);
-    return window.location.hash ? `${trimmedUrl}/#` : trimmedUrl;
+    return window.location.hash ? `${trimmedUrl}#` : trimmedUrl;
```

Comment:
I think the change as is has too much potential to break existing behavior that some apps could depend on. What we could probably get away with is preserving the existing behavior in most cases, but avoiding issues with query parameters by manipulating the hostUrlObject's `path` and `hash` properties to enforce that the path ends in `/` and the `hash` is set as `#` if not present. 

We should also add a story for the next major release to determine if we want to preserve this behavior at all.